lib/puppet/provider/user/user_role_add.rb

```diff
@@ -26,8 +27,22 @@
     value !~ /\s/
   end
 
+  def shell=(value)
```

Comment:
After sitting on it a bit i like the explicitness of this change. +1 to leaving as-is.

@shawnferry if you want to squash commits to per-ticket I think this is good to go